File: Assignment2/ClientPython/model/calendar_event_dto.py
class CalendarEventDTO:
    def __init__(self, name, description, scheduled_time):
        self.name = name
        self.description = description
        self.scheduled_time = scheduled_time

    def __repr__(self):
        return "<CalendarEventDTO(name='%s', description='%s', scheduled_time='%s')>" % (
            self.name, self.description, str(self.scheduled_time))

File: Assignment3/ClientPhp/CalendarEventDto.php
<?php

require_once("CalendarEvent.php");

class CalendarEventDto
{
    public $name;
    public $description;
    public $scheduled_time;

    public static function fromCalendarEvent($calendarEvent)
    {
        return CalendarEventDto::fromValues(
            $calendarEvent->name,
            $calendarEvent->description,
            $calendarEvent->scheduled_time
        );
    }

    public static function fromValues($name, $description, $scheduled_time)
    {
        $calendarEventDto = new CalendarEventDto();
        $calendarEventDto->name = $name;
        $calendarEventDto->description = $description;
        $calendarEventDto->scheduled_time = $scheduled_time;

        return $calendarEventDto;
    }
}
